### PR TITLE
Fix/aws calls

### PR DESCRIPTION
**Pull request by kainoaseto**
@Glympse/devops @AndrewLader PTAL

Just some simple changes to critical AWS calls that will allow their retry system to do incremental backoffs until they succeed. These calls must work for cluster scaling to function.

### CODE REVIEW

File: client/nomad.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func (c *nomadClient) LeastAllocatedNode(capacity *structs.ClusterCapacity,
 // will be assigned and existing allocations will be migrated.
 func (c *nomadClient) DrainNode(nodeID string) (err error) {
 	// Initiate allocation draining for specified node.
-	_, err = c.nomad.Nodes().ToggleDrain(nodeID, true, &nomad.WriteOptions{})
+	_, err = c.nomad.Nodes().ToggleDrain(nodeID, false, nil)
```
Comment (aedades) [inline client/nomad.go]: `false` here won't drain the nodes?

`ToggleDrain()` definition:
```
func (n *Nodes) ToggleDrain(nodeID string, drain bool, q *WriteOptions) (*WriteMeta, error) {
--
  | drainArg := strconv.FormatBool(drain)
  | wm, err := n.client.write("/v1/node/"+nodeID+"/drain?enable="+drainArg, nil, nil, q)
  | if err != nil {
  | return nil, err
  | }
  | return wm, nil
  | }
```

Comment (kainoaseto) [reply client/nomad.go]: That's what I thought also....I have a feeling there's some interesting bugs going on here with nomad. I'll change it back to true for now but write options should absolutely be nil.

**Comment by kainoaseto** [comment]
I'm still seeing some AWS call errors possibly caused by some race conditions, going to need to investigate this more before feeling comfortable merging this in. In particular the error:
`DEBUG: Send Request autoscaling/DescribeAutoScalingGroups failed, will retry, error RequestError: send request failed` has been reoccurring for awhile for some reason. Possible need to use multiple AWS ASG services to do this.

**Comment by djenriquez** [comment]
@kainoaseto Are we still seeing the debug error you're reported ^? Is it a fatal error or can the process continue past the warning?

**Comment by kainoaseto** [comment]
I have not been seeing that error at all anymore, we can continue but the process cannot continue past that warning. It appears it was related to an AWS problem at the time. One thing to note about the Debug item is that this code currently has debugging enabled, we can turn it off in the future but while this particular feature is still in development I will leave it on for now.

**Review by djenriquez — APPROVED** [review]

**Pull request merged**